File: Dia32/EcuacionesCuadraticas.js
// Ecuaciones Cuadráticas en JavaScript

function cuadraticEquationSolutions(A, B, C) {
    let solution = { x1: 0, x2: 0 }
    if (A === 0) {
        if (B === 0) {
            return C === 0
                ? 'Hay infinitas soluciones'
                : 'No hay soluciones'
        } else {
            return { x: -C / B }
        }
    } else {
        let dSquare = B ** 2 - 4 * A * C
        if (dSquare < 0) {
            return 'Las soluciones son imaginarias'
        } else {
            let d = Math.sqrt(dSquare)
            return { x1: (-B + d) / (2 * A), x2: (-B - d) / (2 * A) }
        }
    }
}

// x^2 - 9x + 20 = 0
console.log(cuadraticEquationSolutions(1, -9, 20))
// {x1: 5, x2: 4}

// x^2 - 4x + 5 = 0
console.log(cuadraticEquationSolutions(1, -4, 5))
// Las soluciones son imaginarias

// 5x - 15 = 0
console.log(cuadraticEquationSolutions(0, 5, -10))
// {x: 2}

// 8 = 0
console.log(cuadraticEquationSolutions(0, 0, 8))
// No hay soluciones

// 0 = 0
console.log(cuadraticEquationSolutions(0, 0, 0))
// Hay infinitas soluciones

File: Dia32/EcuacionesCuadraticas.py
# Ecuaciones Cuadráticas en Python

def cuadraticEquationSolutions(A, B, C):
    if A == 0:
        if B == 0:
            if C == 0:
                return 'Hay infinitas soluciones'
            else:
                return 'No hay soluciones'
        else:
            return 'x = {}'.format(-C/B)
    else:
        d = (B**2 - 4*A*C)**.5
        return 'x1 = {}, x2 = {}'.format((-B + d) / (2 * A), (-B - d) / (2 * A))

# x^2 - 9x + 20 = 0
print(cuadraticEquationSolutions(1, -9, 20))
# x1 = 5.0, x2 = 4.0

# x^2 - 4x + 5 = 0
print(cuadraticEquationSolutions(1, -4, 5))
# x1 = (2+1j), x2 = (2-1j)

# 5x - 15 = 0
print(cuadraticEquationSolutions(0, 5, -10))
# x = 2.0

# 8 = 0
print(cuadraticEquationSolutions(0, 0, 8))
# No hay soluciones

# 0 = 0
print(cuadraticEquationSolutions(0, 0, 0))
# Hay infinitas soluciones